### ISSUE TITLE
Error Starting Graylog after upgrade from 3.3.8 to 4.0.0

### ISSUE THREAD

**Issue by chalfling** [Graylog2/graylog2-server#9521]
## Current Behavior

I upgraded Graylog from 3.3.8 to 4.0.0 and started it's service, but got this errors:
https://pastebin.com/WAjVbkY6

I'm stuck at "boot loop", these log lines are repeated indefinitely

## Your Environment

* Graylog Version: 4.0.0
* Java Version: OpenJDK 1.8.0_252
* Elasticsearch Version: 6.8.7
* MongoDB Version: 4.2
* Operating System: FreeBSD 12.1

**Comment by dennisoelkers** [comment]
Hey @chalfling!

Thanks for reporting this. Can you paste the Graylog config file you are using? (Please remove all sensitive details like hostnames/passwords/etc.) That would help us a lot reproducing this issue.

**Comment by mpfz0r** [comment]
```2020-11-17 13:18:33,727 WARN : org.graylog2.shared.security.tls.DefaultTLSProtocolProvider - JRE doesn't support all default TLS protocols. Changing <[TLSv1.2, TLSv1.3]> to <[TLSv1.2]>
2020-11-17 13:18:34,506 INFO : org.graylog2.bootstrap.CmdLineTool - Loaded plugin: AWS plugins 4.0.0 [org.graylog.aws.AWSPlugin]
2020-11-17 13:18:34,507 INFO : org.graylog2.bootstrap.CmdLineTool - Loaded plugin: Integrations 4.0.0 [org.graylog.integrations.IntegrationsPlugin]
2020-11-17 13:18:34,509 INFO : org.graylog2.bootstrap.CmdLineTool - Loaded plugin: Collector 4.0.0 [org.graylog.plugins.collector.CollectorPlugin]
2020-11-17 13:18:34,510 INFO : org.graylog2.bootstrap.CmdLineTool - Loaded plugin: Threat Intelligence Plugin 4.0.0 [org.graylog.plugins.threatintel.ThreatIntelPlugin]
2020-11-17 13:18:35,072 INFO : org.graylog2.bootstrap.CmdLineTool - Running with JVM arguments: -Djava.awt.headless=true -Djava.net.preferIPv4Stack=true -Dapp=graylog -Dlog4j.configurationFile=/usr/local/etc/graylog/log4j2.xml -Xms256m -Xmx1g -XX:-OmitStackTraceInFastThrow -XX:+AggressiveOpts -XX:+CMSClassUnloadingEnabled -XX:+CMSConcurrentMTEnabled -XX:+CMSParallelRemarkEnabled -XX:+DisableExplicitGC -XX:+HeapDumpOnOutOfMemoryError -XX:+ResizeTLAB -XX:+UseCMSInitiatingOccupancyOnly -XX:+UseCompressedOops -XX:+UseConcMarkSweepGC -XX:+UseFastAccessorMethods -XX:+UseParNewGC -XX:CMSInitiatingOccupancyFraction=75 -XX:NewRatio=1
2020-11-17 13:18:35,820 INFO : org.hibernate.validator.internal.util.Version - HV000001: Hibernate Validator null
2020-11-17 13:18:40,741 ERROR: org.graylog2.bootstrap.CmdLineTool - Guice error (more detail on log level debug): No implementation for java.util.Map<org.graylog2.plugin.Version, javax.inject.Provider<org.graylog.events.search.MoreSearchAdapter>> was bound.
  Did you mean?
    org.graylog2.plugin.Version annotated with @com.google.inject.name.Named(value=elasticsearch_version) bound  at com.github.joschi.jadconfig.guice.NamedConfigParametersModule.registerParameters(NamedConfigParametersModule.java:80)
```

It looks like you're missing the `graylog-storage-elasticsearch` plugins..
What's the content of your `plugin` directory?

**Comment by dennisoelkers** [comment]
Hey @chalfling,

@marco is correct. Can you tell us how you upgraded to Graylog 4.0? It looks like some parts are missing.

**Comment by chalfling** [comment]
> ```
> 2020-11-17 13:18:34,506 INFO : org.graylog2.bootstrap.CmdLineTool - Loaded plugin: AWS plugins 4.0.0 [org.graylog.aws.AWSPlugin]
> 2020-11-17 13:18:34,507 INFO : org.graylog2.bootstrap.CmdLineTool - Loaded plugin: Integrations 4.0.0 [org.graylog.integrations.IntegrationsPlugin]
> 2020-11-17 13:18:34,509 INFO : org.graylog2.bootstrap.CmdLineTool - Loaded plugin: Collector 4.0.0 [org.graylog.plugins.collector.CollectorPlugin]
> 2020-11-17 13:18:34,510 INFO : org.graylog2.bootstrap.CmdLineTool - Loaded plugin: Threat Intelligence Plugin 4.0.0 [org.graylog.plugins.threatintel.ThreatIntelPlugin]
> 2020-11-17 13:18:35,072 INFO : org.graylog2.bootstrap.CmdLineTool - Running with JVM arguments: -Djava.awt.headless=true -Djava.net.preferIPv4Stack=true -Dapp=graylog -Dlog4j.configurationFile=/usr/local/etc/graylog/log4j2.xml -Xms256m -Xmx1g -XX:-OmitStackTraceInFastThrow -XX:+AggressiveOpts -XX:+CMSClassUnloadingEnabled -XX:+CMSConcurrentMTEnabled -XX:+CMSParallelRemarkEnabled -XX:+DisableExplicitGC -XX:+HeapDumpOnOutOfMemoryError -XX:+ResizeTLAB -XX:+UseCMSInitiatingOccupancyOnly -XX:+UseCompressedOops -XX:+UseConcMarkSweepGC -XX:+UseFastAccessorMethods -XX:+UseParNewGC -XX:CMSInitiatingOccupancyFraction=75 -XX:NewRatio=1
> 2020-11-17 13:18:35,820 INFO : org.hibernate.validator.internal.util.Version - HV000001: Hibernate Validator null
> 2020-11-17 13:18:40,741 ERROR: org.graylog2.bootstrap.CmdLineTool - Guice error (more detail on log level debug): No implementation for java.util.Map<org.graylog2.plugin.Version, javax.inject.Provider<org.graylog.events.search.MoreSearchAdapter>> was bound.
>   Did you mean?
>     org.graylog2.plugin.Version annotated with @com.google.inject.name.Named(value=elasticsearch_version) bound  at com.github.joschi.jadconfig.guice.NamedConfigParametersModule.registerParameters(NamedConfigParametersModule.java:80)
> ```
> 
> It looks like you're missing the `graylog-storage-elasticsearch` plugins..
> What's the content of your `plugin` directory?

ls /usr/local/share/graylog/plugin/
graylog-plugin-aws-4.0.0.jar           graylog-plugin-collector-4.0.0.jar     graylog-plugin-integrations-4.0.0.jar  graylog-plugin-threatintel-4.0.0.jar

**Comment by chalfling** [comment]
> Hey @chalfling,
> 
> @marco is correct. Can you tell us how you upgraded to Graylog 4.0? It looks like some parts are missing.
Hey @dennisoelkers I use same FreeBSD ports from 3.3.8 and only changed version 4.0.0. I'll copy all plugins from folder and try again. If works I'll update FreeBSD ports.

Thank you!

**Comment by chalfling** [comment]
Thank you @mpfz0r and @dennisoelkers !

I changed FreeBSD ports. You can first look in: https://reviews.freebsd.org/D27253
Waiting for aprove from Maintainer dch.

**Comment by dennisoelkers** [comment]
Thanks a lot for the update, @chalfling! Good work and thanks for updating the FreeBSD port!

**Comment by Ghostbird** [comment]
I'm receiving this error too, on a Ubuntu 20 server. Turns out, I had mounted my entire `plugins` directory in the docker compose, which is convenient when adding new plugins, and apparently works without issue in graylog 3. In graylog 4 it crashes the container.
This issue put me on the right track, as I realised that my container could not find some plugins that should be present in the container by default.
